### ISSUE TITLE
negative row index

### ISSUE THREAD

**Issue by kimim** [techascent/tech.ml.dataset#177]
Is it possible to have negative row index in `select`, for example:

```clojure
(ds/row-count DS) ;=> 100
(ds/select DS :all [0 1 2 3 -3 -2 -1])
```

will become:

```clojure
(ds/select DS :all [0 1 2 3 97 98 99])
```

**Comment by kimim** [comment]
Furthermore, how about select both column and row with index-seq(including negative):

```clojure
(defn select-v2 [dataset col-index row-index]
  (let [make-pos (fn [total x] (if (neg? x) (+ x total) x))
        col-total (ds/column-count dataset)
        colname-seq (->> (map (partial make-pos col-total) col-index)
                         (map #(nth (ds/column-names dataset) %)))
        row-total (ds/row-count dataset)
        row-index (map (partial make-pos row-total) row-index)]
    (ds/select dataset colname-seq row-index)))

(select-v2 train-data [0 1 2 3 -3 -2 -1] [0 1 2 3 4 -3 -2 -1])
```

**Comment by cnuernber** [comment]
Certainly possible as you show above. I also think it is clear to me that this would be useful shorthand.  I have also wanted to just select columns with indexes.  My only hesitation is that this does add a preprocessing step to the index generation so it would have to be done outside of critical paths.

**Comment by kimim** [comment]
Yes. That will introduce a little overhead. If you decide to add below interface, I'd like to create a PR.

```clojure
(select-columns-with-index DS [0 1 2 3 -3 -2 -1])
(select-rows-with-index DS [0 1 2 3 -3 -2 -1])
(select-with-index DS [0 1 2 3 -3 -2 -1] [0 1 2 3 -3 -2 -1])
```

**Comment by cnuernber** [comment]
I would appreciate a PR, the interface looks fine to me, thank you.  This will allow users to experiment with it.  For your 'map' operation to adjust indexes, [dtype-next/emap](https://cnuernber.github.io/dtype-next/tech.v3.datatype.html#var-emap) with :int64 datatype preserves the ability of the system to parallelize access to the result if the input is random access like in your examples above or a clojure `range`.  `clojure.core/map` creates a pure sequence which will force the system to copy the result into either an integer or long buffer as the columns have to have randomly access in their index buffers.

**Comment by kimim** [comment]
I change `with` to `by`, in order to make the function a bit shorter.

**Comment by cnuernber** [comment]
Fixed by #179
